detections/application/splunk_protocol_impersonation_weak_encryption_configuration.yml

```diff
@@ -0,0 +1,50 @@
+name: Splunk Protocol Impersonation Weak Encryption Configuration
+id: 900892bf-70a9-4787-8c99-546dd98ce461
+version: 1
+date: '2022-05-25'
+author: Lou Stella, Splunk
+type: Hunting
+datamodel: []
+description: On June 14th, 2022, Splunk released a security advisory relating to TLS validation occuring within the httplib and urllib python libraries shipped with Splunk. In addition to upgrading to Splunk Enterprise 9.0 or later, several configuration settings need to be set. This search will check those configurations on the search head it is run from as well as its search peers. In addition to these settings, the PYTHONHTTPSVERIFY setting in $SPLUNK_HOME/etc/splunk-launch.conf needs to be enabled as well. Other components such as additional search heads or anything this rest command cannot be distributed to will need to be manually checked.
+search: '| rest /services/server/info | table splunk_server version server_roles | join splunk_server [| rest /servicesNS/nobody/search/configs/conf-server/ search="PythonSslClientConfig" | table splunk_server sslVerifyServerCert sslVerifyServerName] | join splunk_server [| rest /servicesNS/nobody/search/configs/conf-web/settings |  table splunk_server serverCert sslVersions] | rename sslVerifyServerCert as "Server.conf:PythonSSLClientConfig:sslVerifyServerCert", sslVerifyServerName as "Server.conf:PythonSSLClientConfig:sslVerifyServerName", serverCert as "Web.conf:Settings:serverCert", sslVersions as "Web.conf:Settings:sslVersions" | `splunk_protocol_impersonation_weak_encryption_configuration_filter`'
+how_to_implement: The user running this search is required to have a permission allowing them to dispatch REST requests to indexers. 
+known_false_positives: While all of the settings on each device returned by this search may appear to be hardened, you will still need to verify the value of PYTHONHTTPSVERIFY in $SPLUNK_HOME/etc/splunk-launch.conf on each device in order to harden the python configuration.
+references:
+- https://docs.splunk.com/Documentation/Splunk/9.0.0/Security/EnableTLSCertHostnameValidation
+- https://www.splunk.com/en_us/product-security/announcements/svd-2022-0601.html
+tags:
+  analytic_story:
+  - Splunk Vulnerabilities
+  asset_type: Endpoint
+  cis20:
+  - CIS 3
+  - CIS 5
+  - CIS 16
+  cve:
+  - CVE-2022-32151 
+  confidence: 100
+  context:
+  - Source:Endpoint
+  dataset:
+  - https://media.githubusercontent.com/media/splunk/attack_data/master/datasets/attack_techniques/T1213/audittrail/audittrail.log
+  impact: 50
+  kill_chain_phases:
+  - Exploitation
+  message: $splunk_server$ may not be properly validating TLS Certificates
+  mitre_attack_id:
+  - T1001.003
+  nist:
+  - DE.CM
+  observable:
+  - name: splunk_server
+    type: Hostname
+    role:
+    - Victim
+  product:
+  - Splunk Enterprise
+  - Splunk Enterprise Security
+  - Splunk Cloud
+  required_fields:
```

Comment:
shouldn't we have required fields here?

Reply:
`required_fields` is for fields that are required to be in the logs we use, right? If this search doesn't use any logs, it doesn't really require any fields to be in logs.